src/app/auth/guards/auth.guard.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+import { AccountService } from '@shared/services/account/account.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+  constructor(
+    private account: AccountService,
+    private router: Router,
+  ) {}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    if (this.account.getAccount()?.accountId) {
+      this.account.refreshArchives().then((archives) => {
+        const ownArchives = archives.filter(
+          (archive) => !archive.isPending()
+        );
+        if (ownArchives.length > 0) {
```

Comment:
Ah, this is the second time I've seen this snippet of code! It appears to be used for the same purpose; maybe my above comment about extracting a helper should lead to a more widely-visible helper than I was originally imagining.